### PR TITLE
Add jwt secret handling action

### PR DESCRIPTION
**Pull request by stdevMac**
## Changes:
- Use newed created action for Jwt secret handling

## Types of changes
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (no functional changes, no api changes)

## Testing
**Requires testing**

- [x] Yes
- [ ] No

**In case you checked yes, did you write tests?**

- [x] Yes
- [ ] No

### CODE REVIEW

File: cli/actions/jwt_secrets.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+type CreateJwtSecretOptions struct {
+	JwtPath        string
+	Network        string
+	GenerationPath string
+}
+
+func (s *sedgeActions) CreateJwtSecrets(options CreateJwtSecretOptions) (string, error) {
+	if options.Network == "" {
+		return "", fmt.Errorf("network not found")
```
Comment (AntiD2ta) [inline cli/actions/jwt_secrets.go:20]: Put error message in `actions/errors.go` (yeah, for the Actions is different, @adriantpaez introduced this and we are now respecting it)

File: cli/actions/jwt_secrets_test.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+			sedgeActions := actions.NewSedgeActions(nil, nil, nil)
+			jwtPath, err := sedgeActions.CreateJwtSecrets(tc.options)
+			if err != nil {
+				if !tc.err {
+					t.Errorf("Got: %v, want error: %v", err, tc.err)
+				}
+				return
+			}
+			if tc.err || tc.options.JwtPath != "" {
+				return
```
Comment (AntiD2ta) [inline cli/actions/jwt_secrets_test.go:52]: I believe that if `tc.error` is true and `err` is nil then the test failed. If I am correct then please replace the `return` with a `t.Error(...)` and move `tc.options.JwtPath != ""` to another `if` statement

File: cli/cli.go [hunk 3]
```diff
@@ -257,14 +256,14 @@ func runCliCmd(cmd *cobra.Command, args []string, flags *CliCmdFlags, clientImag
...
-			return []error{err}
-		}
+	jwtPath := flags.jwtPath
+	jwtPath, err = sedgeActions.CreateJwtSecrets(actions.CreateJwtSecretOptions{
+		JwtPath:        jwtPath,
+		Network:        flags.network,
+		GenerationPath: flags.generationPath,
+	})
+	if err != nil {
+		return nil
```
Comment (AntiD2ta) [inline cli/cli.go:266]:
```suggestion
		return []error{err}
```

**Review by AntiD2ta — CHANGES_REQUESTED** [review]
2 requested changes and 1 concern to be addressed

File: cli/cli.go (hunk 3 above)
Comment (stdevMac) [reply cli/cli.go]: Done at [5c417b9](https://github.com/NethermindEth/sedge/pull/210/commits/5c417b929b8888c9623fc674c2c4b1ba36ad8805)

File: cli/actions/jwt_secrets_test.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
+package actions_test
+
+import (
+	"github.com/NethermindEth/sedge/cli/actions"
+	"github.com/stretchr/testify/assert"
+	"path/filepath"
+	"testing"
+)
+
+func TestSedgeActions_CreateJwtSecretsContainers(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (stdevMac) [reply cli/actions/jwt_secrets_test.go]: Done at [e230f06](https://github.com/NethermindEth/sedge/pull/210/commits/e230f06ece46b87567367c922e2571885709abbb)

File: cli/actions/jwt_secrets_test.go (hunk 2 above)
Comment (stdevMac) [reply cli/actions/jwt_secrets_test.go:53]: You were right, done at [e83617d](https://github.com/NethermindEth/sedge/pull/210/commits/e83617d10d7504004f476ab31542064e7d143786)

File: CHANGELOG.md [hunk 5]
```diff
@@ -16,6 +16,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+- Use newed created action for Jwt secret handling
```
Comment (adriantpaez) [inline CHANGELOG.md:19]:
```suggestion
- Use newed created action for JWT secret handling
```

File: cli/actions/actions.go [hunk 6]
```diff
@@ -27,6 +27,7 @@ type SedgeActions interface {
 	SetupContainers(SetupContainersOptions) error
 	RunContainers(RunContainersOptions) error
 	InstallDependencies(InstallDependenciesOptions) error
+	CreateJwtSecrets(CreateJwtSecretOptions) (string, error)
```
Comment (adriantpaez) [inline cli/actions/actions.go:30]:
```suggestion
	CreateJWTSecrets(CreateJwtSecretOptions) (string, error)
```
Plz, review the rest of the code to replace Jwt for JWT

File: cli/actions/errors.go [hunk 7]
```diff
@@ -18,3 +18,5 @@ package actions
 import "errors"
 
 var ErrUnsupportedValidatorClient = errors.New("unsupported validator client")
+
+var ErrorNetworkNotFound = errors.New("network not found")
```
Comment (adriantpaez) [inline cli/actions/errors.go:22]:
```suggestion
var (
	ErrUnsupportedValidatorClient = errors.New("unsupported validator client")
	ErrorNetworkNotFound          = errors.New("network not found")
)
```

**Review by adriantpaez — CHANGES_REQUESTED** [review]

File: cli/actions/jwt_secrets_test.go [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			sedgeActions := actions.NewSedgeActions(nil, nil, nil)
+			jwtPath, err := sedgeActions.CreateJwtSecrets(tc.options)
+			if err != nil {
+				assert.True(t, tc.err)
+				return
+			}
```
[the remark this answers is not in the document]
Comment (stdevMac) [reply cli/actions/jwt_secrets_test.go]: done [6180380](https://github.com/NethermindEth/sedge/pull/210/commits/6180380645a004a14502630ff40f8b0bc2a8d460)

File: cli/actions/errors.go (hunk 7 above)
Comment (stdevMac) [reply cli/actions/errors.go]: done [6180380](https://github.com/NethermindEth/sedge/pull/210/commits/6180380645a004a14502630ff40f8b0bc2a8d460)

**Review by AntiD2ta — APPROVED** [review]

**Pull request merged**